### PR TITLE
Add topologySpreadConstraints to jitsu helm chart

### PR DESCRIPTION
**Pull request by mleklund**

### CODE REVIEW

**Comment by hendrikheil** [comment]
LGTM, @echozio wby?

File: values.yaml [hunk 1]
```diff
@@ -262,6 +262,7 @@ console:
   nodeSelector: {}
   tolerations: []
   affinity: {}
+  topologySpreadConstraints: ""
```
Comment (echozio) [inline values.yaml:265]: Wouldn't it make more sense to use a list? Or is there some use case here where templating is particularly useful?

Comment (mleklund) [reply values.yaml:265]: Maybe, and I am certainly open to that.  Using a tpl function allows something like this:

```
  topologySpreadConstraints: |
    - maxSkew: 1
      topologyKey: "topology.kubernetes.io/zone"
      whenUnsatisfiable: DoNotSchedule
      labelSelector:
        matchLabels:
          {{- include "jitsu.selectorLabels" . | nindent 10 }}
          {{- include "jitsu.ingest.selectorLabels" . | nindent 10 }}
    - maxSkew: 1
      topologyKey: "kubernetes.io/hostname"
      whenUnsatisfiable: DoNotSchedule
      labelSelector:
        matchLabels:
          {{- include "jitsu.selectorLabels" . | nindent 10 }}
          {{- include "jitsu.ingest.selectorLabels" . | nindent 10 }}
```

where you don't have to know as much about the labels.

File: templates/syncctl/deployment.yaml [hunk 2]
```diff
@@ -104,3 +104,5 @@ spec:
       tolerations:
         {{- toYaml . | nindent 8 }}
       {{- end }}
+      topologySpreadConstraints:
+        {{- tpl .Values.syncctl.topologySpreadConstraints . | nindent 8 }}
```
Comment (echozio) [inline templates/syncctl/deployment.yaml:108]: What do you think about doing this in order to support both a template and a list?
```suggestion
      {{- with .Values.syncctl.topologySpreadConstraints }}
      topologySpreadConstraints:
        {{- if kindIs "string" . }}
        {{- tpl . $ | nindent 8 }}
        {{- else }}
        {{- toYaml . | nindent 8 }}
        {{- end }}
      {{- end }}
```

File: values.yaml (hunk 1 above)
Comment (echozio) [reply values.yaml:265]: I agree that's worth supporting. See my other comment for a suggestion on how to support both.

**Comment by mleklund** [comment]
Agreed, it is a more flexible and elegant solution.  Updated with a commit.

**Review by echozio — APPROVED** [review]
Looks great! Thanks for the contribution

**Pull request merged**